### DISCUSSION TITLE
How to train a FLUX lora?

### DISCUSSION THREAD

**Discussion by Xuan-World** [huggingface/diffusers#10061] [Q&A]
I don't want to train a dreambooth lora or something like that. I just want to train a lora on FLUX-schnell. So which script should I refer to in ''examples'' directory?

**Comment by bmaxdk** [comment]
[train_text_to_image_lora.py](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_lora.py) in the `examples` directory, will be the one you looking for since it is designed specifically for training LoRA models without involving DreamBooth.


```py
python train_text_to_image_lora.py \
  --pretrained_model_name_or_path="path_or_identifier_to_FLUX-schnell" \ # Path or Hugging Face identifier for FLUX-schnell
  --train_data_dir=".......
  # add other cofig

```

**Reply by Xuan-World** [reply]
thanks a lot.

**Reply by Doby-Xu** [reply]
Hey guys. Have you ever tried to run that script? I don't think that would work, since the implementation in the example is completely based on Stabel Diffusion, instead of FLUX.

**Reply by Xuan-World** [reply]
Yes, it's really tough to change that script to adapt to FLUX. I'm trying on it for several days. A Suggestion: refer to the implementation script of DreamBooth FLUX LoRA and Control-FLUX LoRA.

**Reply by asomoza** [reply]
you will have an easier time with the other way around: start with the dreamBooth FLUX LoRA script  and remove all the parts that are for dreambooth.
